### DISCUSSION TITLE
GUI対応について

### DISCUSSION THREAD

**Discussion by emadurandal** [Qithub-BOT/QiiCipher#37] [アイデア]
実は、先程ちょいとこんなの作りまして。

https://user-images.githubusercontent.com/3821214/120333242-75f58700-c32a-11eb-94e3-55b5fdeb96fa.mp4

.batファイルとPowershellコードを組み合わせた簡素なものではありますが、Windowsのエクスプローラーからドラッグ＆ドロップで利用できる感じです。
暗号化するのにちょっと大きなサイズのファイルが必要だと思って、試しに３D系のファイル（42.5 MB）でやってみました。暗号化＆復号しても元通り使えることをデモしています。

発想的には以前Qiitaでリンクいただいた以下の記事と同じです。
https://qiita.com/emadurandal/items/b13ee6a002fa19fb82d7

GUI対応は色々と方針というか、皆様と話し合って進めていくのがいいのかな、と思っているのと、まだコードも荒削りなのでまだPRは作らずにおきます（別件のPRの修正もありますし💦）

あとは、署名と検証もそれぞれarchive.shとdearchive.shでやるようにすると言うことなしかも笑

復号時の秘密鍵については、環境変数で場所を示していれば（図中のQIICIPHER_GITHUB_KEYNAME）、いちいちダイアログで聞かずに環境変数の方を見てくれるようにしています。
（他、.bat->powershell->wsl呼び出しだとPATH環境変数が普通にやるとうまく継承されないっぽく、現状はQIICHIPHER_BINでbinフォルダを教えてあげている感じです）
![image](https://user-images.githubusercontent.com/3821214/120333878-1a77c900-c32b-11eb-8a0b-308475f11975.png)

似たような感じでMacとかも対応できそうですね（.commandとか、あとはAutomatorでもいいかも）

**Comment by KEINOS** [comment]
すごーい！！GUI GUI きますね！w

やるやる詐欺のテストの実装が出来たら、`brew` で macOS や Linux でも CUI インストールできたらいいなと思っています。
その時、デスクトップとかに `qiicipher.command` を作成できたら、ドラッグ & ドロップで使えそうですね。

Win の `choco` でもインストール可能になったら `chocolatey install qiicipher-gui` とかで GUI 版をインストールできたら格好いい！！

GUI の方針はあるようで、まだないので、この Issue #32 は Issue ではなくディスカッション #37 に移したいと思います。

**Comment by emadurandal** [comment]
> 発想的には以前Qiitaでリンクいただいた以下の記事と同じです。
> https://qiita.com/emadurandal/items/b13ee6a002fa19fb82d7

これ、会社の人から「.batって人によっては警戒するから」っていうフィードバックがあったりもしました。なので、何を採用するかはよく考えた方がよさそうですね。
うちの社内はとりあえずこれで暫定運用していくかもです。

アプリ形式でbrewとかchocoとかscoopでサクッとWSL側のセットアップも全部できちゃうと最高そうですね。

**Comment by KEINOS** [comment]
> .batって人によっては警戒するから

確かにそうですよね。すべての `bat` がいいとは限らないので、誰ともわからんものは「基本怪しいと考える」のは正しい判断だと思います。

<details><summary>こういう感じ</summary>

<div>「正義の味方で〜っしゅ」</div>
<img src="https://i2-prod.birminghammail.co.uk/news/weird-news/article1717684.ece/ALTERNATES/s1200b/batman.jpg" width="100%" />

</details>
